client/src/pages/Discovery/Discovery.tsx

```diff
@@ -0,0 +1,140 @@
+import { useEffect, useState } from 'react';
+import { useAuth } from '../../context/useAuthContext';
+import { Contest } from '../../interface/User';
+import { Column } from '../../interface/Discovery';
+import { getAllContests } from '../../helpers/APICalls/contest';
+import AuthHeader from '../../components/AuthHeader/AuthHeader';
+import Grid from '@material-ui/core/Grid';
+import Typography from '@material-ui/core/Typography';
+import CircularProgress from '@material-ui/core/CircularProgress';
+import Container from '@material-ui/core/Container';
+import Button from '@material-ui/core/Button';
+import Paper from '@material-ui/core/Paper';
+import TableContainer from '@material-ui/core/TableContainer';
+import Table from '@material-ui/core/Table';
+import TableHead from '@material-ui/core/TableHead';
+import TableRow from '@material-ui/core/TableRow';
+import TableCell from '@material-ui/core/TableCell';
+import TableBody from '@material-ui/core/TableBody';
+import TablePagination from '@material-ui/core/TablePagination';
+import useStyles from './useStyles';
+
+export default function Discovery(): JSX.Element {
+    const [contests, setContests] = useState<Contest[]>([]);
+    const [page, setPage] = useState(0);
+    const [rowsPerPage, setRowsPerPage] = useState(10)
+    const { loggedInUser } = useAuth();
+    const classes = useStyles();
+
+    useEffect(() => {
+        async function getAll() {
+            const allContests = await getAllContests();
+            if (allContests.contests) {
+                setContests(allContests.contests)
+            } else {
+                console.log("No Contests Found")
+                new Error("Could Not Get Contests")
+            }
+        };
+        getAll()
+    }, []);
+
+
+
+    const columns: Column[] = [
```

Comment:
You can add any styles to the cell using a class and hardcode labels within the cells. I don't think we will need this array of objects :)

Reply:
had the weirdest thing happening when I was trying to map through columns, as you can see I had to hard code the headings, was initially going to map through but for some reason it was not recognizing the columns variable in the return. will update code and push shortly